app/jobs/notify_job.rb

```diff
@@ -5,18 +5,29 @@ class NotifyJob < ApplicationJob
 
   def perform(notification_id:)
     notification = Notification.kept.includes(:subscription).find(notification_id)
-
     return unless notification.subscription.enabled?
 
     data = ActiveModelSerializers::Adapter.create(NotificationSerializer.new(notification)).to_json
     hmac = Encryptor.hmac(notification.subscription.secret, data)
-
-    response = client.post(notification.subscription.callback_url, data, 'PECS-SIGNATURE': hmac, 'PECS-NOTIFICATION-ID': notification_id)
-
-    notification.delivered_at = DateTime.now if response.success?
+    success = false # assume failure
```

Comment:
do we need this `success` var?
It sounds a bit redundant (I might be wrong) since `response.success?` could be used for it,
`raise 'Notification failed' unless success` could be put inside the `else`
and maybe rescue should raise a different kind of exception.

Reply:
@alexdesi thanks for looking - the tricky thing is that I want to do raise an exception _after_ I have updated the notification record, so the status needs to be stored somewhere and the response object might not always exist. I'll have another go at making this cleaner...

Reply:
i've removed the `success` variable; however I don't think the shape of the logic can be much improved ?

Reply:
I agree, that code does a few things.
Nitpick: if you wish you can remove the `response.present?` check
and use the safe navigation  (&.):
`raise 'Notification failed' unless response&.success?`

I noticed it's a common practice in our project.
Up to you.